### PR TITLE
Feature/21 implement windowmanagement module

### PR DESCRIPTION
**Pull request by limingyialpha**
Please test the functionality and review the code.
No configuration parameter needed to test

### CODE REVIEW

**Comment by sjend** [comment]
Could you rebase this onto the current develop and add `NotifyOnEnqueueFinished` in your producers' `StopCapture` methods?

**Comment by Gordin508** [comment]
When you rebase onto `develop` now, keep in mind that `develop` now already contains the additions in the GlobalHook (WM_ENTERSIZEMOVE and WM_EXITSIZEMOVE).

File: application/Common/Shared/Utility/Utility.cs [hunk 1]
```diff
@@ -50,5 +53,45 @@ public static void SetAndDispatch(ref bool variable, bool value, Action onTrue,
...
+        ///     Get the title of a window from it's Hwnd
+        /// </summary>
+        /// <param name="hwnd"> Hwnd of the window</param>
+        /// <returns>the title of the window in string</returns>
+        public static string GetWindowTitleFromHwnd(IntPtr hwnd)
+        {
+            int length1 = NativeMethods.GetWindowTextLength(hwnd);
+            StringBuilder sb1 = new StringBuilder(length1 + 1);
+            NativeMethods.GetWindowText(hwnd, sb1, sb1.Capacity);
+            return sb1.ToString();
```
Comment (sjend) [inline application/Common/Shared/Utility/Utility.cs]: The variable names here are not ideal - I would suggest something like:
```
var textLength = NativeMethods.GetWindowTextLength(hwnd);
var windowTextStringBuilder = new StringBuilder(textLength + 1);
NativeMethods.GetWindowText(hwnd, windowTextStringBuilder, windowTextStringBuilder.Capacity);
return windowTextStringBuilder.ToString();
```

File: application/Common/Shared/Utility/Utility.cs [hunk 2]
```diff
@@ -50,5 +53,45 @@ public static void SetAndDispatch(ref bool variable, bool value, Action onTrue,
...
+        /// <summary>
+        ///     Get the The name of the process associated with a window
+        ///     from the window's Hwnd
+        /// </summary>
+        /// <param name="hwnd">Hwnd of the window</param>
+        /// <returns>the name of the process associated with the window</returns>
+        public static string GetProcessNameFromHwnd(IntPtr hwnd)
+        {
+            NativeMethods.GetWindowThreadProcessId(hwnd, out var pid);
+            Process process = Process.GetProcessById((int)pid);
```
Comment (sjend) [inline application/Common/Shared/Utility/Utility.cs]: This could be `var process`

File: application/Common/Shared/Utility/Utility.cs [hunk 3]
```diff
@@ -50,5 +53,45 @@ public static void SetAndDispatch(ref bool variable, bool value, Action onTrue,
...
+            return process.ProcessName;
+        }
+
+        public static bool IsRectSizeEqual(Rectangle rectA, Rectangle rectB) 
+        {
+            return GetWindowWidth(rectA) == GetWindowWidth(rectB) && GetWindowHeight(rectA) == GetWindowHeight(rectB);
+        }
+
+        public static int GetWindowWidth(Rectangle rect) {
```
Comment (sjend) [inline application/Common/Shared/Utility/Utility.cs]: Could you move the "{" to the next line?

**Pull request merged**